File: vocalmind_2/vocalmind/types/onboarding.ts
// ─────────────────────────────────────────────
// 온보딩 타입 — 상담 위저드, 긴장 분석 결과
// ─────────────────────────────────────────────

// ── 온보딩 (Onboarding) ──

export interface OnboardingTension {
  overall: number;
  laryngeal: number;
  tongue_root: number;
  jaw: number;
  register_break: number;
  detail: string;
}

// TensionResult alias (하위 호환)
export type TensionResult = OnboardingTension;

export interface OnboardingConsultation {
  problems: string[];
  roadmap: string[];
  suggested_stage_id: number;
  summary: string;
}

export interface OnboardingResult {
  tension: OnboardingTension;
  consultation: OnboardingConsultation;
}

File: vocalmind_2/vocalmind/backend/schemas/onboarding.py
"""온보딩 API 요청/응답 스키마."""
from __future__ import annotations
from pydantic import BaseModel


class OnboardingTensionResponse(BaseModel):
    overall: float
    laryngeal: float
    tongue_root: float
    jaw: float
    register_break: float
    detail: str


class ConsultationResponse(BaseModel):
    problems: list[str]
    roadmap: list[str]
    suggested_stage_id: int
    summary: str


class OnboardingAnalyzeResponse(BaseModel):
    tension: OnboardingTensionResponse
    consultation: ConsultationResponse


class TTSRequest(BaseModel):
    text: str
